### PR TITLE
Fix checkpoint and pretrained-weight loading

### PR DESCRIPTION
**Pull request by pierrebrunelle**
## Summary

Fixes for checkpoint and pretrained-weight loading, surfaced by #11 and found in #13 (credit to @ps-spark for spotting the hyphen bug there first):

- `Yolox.from_pretrained("yolox-s")` 404'd: `get_named_config` normalizes `-` to `_`, but the pretrained-weights filename/URL did not, so the Megvii release asset (`yolox_s.pth`) was never found under the hyphenated name.
- Training checkpoints stored `best_ap`/`curr_ap` as `numpy.float64` whenever an eval produced at least one box (values come straight from `cocoEval.stats`). `torch.load(weights_only=True)`, the default on torch >= 2.6, rejects those on every numpy version. Both values are now cast to builtin `float` in `save_ckpt`.
- New `yolox.utils.load_checkpoint()` always calls `torch.load(weights_only=True)` and never falls back to full unpickling. It allowlists only the three numpy globals that older trainers wrote (`scalar`, `dtype`, `Float64DType`), scoped to the call. `from_pretrained`, `yolox eval`, and trainer `--resume`/`--ckpt` all load through it.

What loads where:

| torch | Megvii weights, new checkpoints | old checkpoints with numpy scalars | pickle payload |
|---|---|---|---|
| 2.2 to 2.4 | yes | `RuntimeError` with a one-line conversion for trusted files | refused |
| 2.5 | yes | yes if written under the same numpy major version | refused |
| >= 2.6 | yes | yes | refused |

#### Test plan

- [x] `tests/utils/test_checkpoint.py`: builtin floats, numpy scalars, `from_pretrained` from a file, and a `__reduce__` payload (with and without numpy nearby) that must not run
- [x] Suite on torch 2.2.2/numpy 1.26/opencv 4.11 and torch 2.3.0/numpy 2.2/opencv 4.13: 5 passed, 1 skipped (numpy-scalar case needs torch >= 2.6), `test_detections` passes
- [x] Suite on torch 2.14/numpy 2.4: checkpoint tests pass; `test_detections` fails on main too (opencv-python >= 4.14 drift, separate issue)
- [x] Load matrix across torch 2.2.2, 2.4.1, 2.5.1, 2.6.0, 2.14 with files written under numpy 1.26 and 2.4
- [x] Pixeltable core against this branch (torch 2.14): `tests/functions/test_yolox.py`, `test_vision.py::test_eval`, `test_vision.py::test_bboxes_draw` (4 passed) and notebooks `img-detect-objects`, `img-visualize-detections`, `object-detection-in-videos` (3 passed)
- [ ] `yolox eval` and trainer resume need CUDA; not run locally

Generated with [Devin](https://devin.ai)

### CODE REVIEW

File: yolox/models/yolox.py [hunk 1]
```diff
@@ -115,12 +116,21 @@ def from_pretrained(
...
         model.training = False
-        weights = torch.load(path, map_location=torch.device(device))
+        try:
+            weights = torch.load(path, map_location=torch.device(device), weights_only=True)
+        except Exception:
+            warnings.warn(
+                f"weights_only=True failed to load {path}; retrying with weights_only=False. "
+                "Only load checkpoints from sources you trust."
+            )
+            weights = torch.load(path, map_location=torch.device(device), weights_only=False)
```
[the remark this answers is not in the document]
Comment (pierrebrunelle) [reply yolox/models/yolox.py]: Confirmed. With the fallback, a checkpoint whose pickle calls `os.system` runs on load; I reproduced it on torch 2.14. It was also a regression, because torch >= 2.6 already defaults to `weights_only=True`. Fixed in b9d600d: the fallback is gone. `load_checkpoint()` in `yolox/utils/checkpoint.py` always uses `weights_only=True`, allowlists only the three numpy globals that older trainers wrote (`scalar`, `dtype`, `Float64DType`), and raises an error with a one-line conversion for files it rejects. `tests/utils/test_checkpoint.py` asserts that a pickle payload is refused and never runs.

File: yolox/cli/eval.py [hunk 2]
```diff
@@ -153,7 +153,14 @@ def eval(config: YoloxConfig, args, num_gpu):
...
         loc = "cuda:{}".format(rank)
-        ckpt = torch.load(ckpt_file, map_location=loc)
+        try:
+            ckpt = torch.load(ckpt_file, map_location=loc, weights_only=True)
+        except Exception:
+            warnings.warn(
+                f"weights_only=True failed to load {ckpt_file}; retrying with weights_only=False. "
+                "Only load checkpoints from sources you trust."
+            )
+            ckpt = torch.load(ckpt_file, map_location=loc, weights_only=False)
```
[the remark this answers is not in the document]
Comment (pierrebrunelle) [reply yolox/cli/eval.py]: Same fix as in `from_pretrained` (b9d600d): `yolox eval` now calls `load_checkpoint()`, which has no unrestricted retry. Trainer `--resume` and `--ckpt` fine-tuning load through it too; before this they called `torch.load` with the torch default.
